x-pack/plugins/security/server/routes/indices/get_fields.ts

```diff
@@ -23,21 +37,41 @@ export function defineGetFieldsRoutes({ router, clusterClient }: RouteDefinition
             fields: '*',
             allowNoIndices: false,
             includeDefaults: true,
-          })) as Record<string, { mappings: Record<string, unknown> }>;
+          })) as FieldMappingResponse;
 
         // The flow is the following (see response format at https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-field-mapping.html):
         // 1. Iterate over all matched indices.
         // 2. Extract all the field names from the `mappings` field of the particular index.
-        // 3. Collect and flatten the list of the field names.
+        // 3. Collect and flatten the list of the field names, omitting any fields without mappings, and any runtime fields
         // 4. Use `Set` to get only unique field names.
+        const fields = Array.from(
+          new Set(
+            Object.values(indexMappings)
+              .map((indexMapping) => {
+                return Object.entries(indexMapping.mappings).map(([fieldName, properties]) => {
+                  const mappingValues = Object.values(properties.mapping);
+                  const hasMapping = mappingValues.length > 0;
+
+                  const isRuntimeField = hasMapping && mappingValues[0]?.type === 'runtime';
+
+                  // fields without mappings are internal fields such as `_routing` and `_index`,
+                  // and therefore don't make sense as autocomplete suggestions for FLS.
+
+                  // Runtime fields are not securable via FLS.
+                  // Administrators should instead secure access to the fields which derive this information.
+                  if (!hasMapping || isRuntimeField) {
+                    return null;
+                  }
+
+                  return fieldName;
+                });
+              })
+              .flat()
+          )
+        ).filter((field) => field !== null) as string[];
```

Comment:
__nit:__ wondering if we can simplify this a bit (to have less iterations)?
```suggestion
        const fields = Array.from(
          new Set(
            Object.values(indexMappings).flatMap((indexMapping) => {
              return Object.keys(indexMapping.mappings).filter((fieldName) => {
                const mappingValues = Object.values(indexMapping.mappings[fieldName].mapping);
                const hasMapping = mappingValues.length > 0;

                const isRuntimeField = hasMapping && mappingValues[0]?.type === 'runtime';

                // fields without mappings are internal fields such as `_routing` and `_index`,
                // and therefore don't make sense as autocomplete suggestions for FLS.

                // Runtime fields are not securable via FLS.
                // Administrators should instead secure access to the fields which derive this information.
                return hasMapping && !isRuntimeField;
              });
            })
          )
        );
```